src/main/java/de/medizininformatikinitiative/torch/cql/CqlClient.java

```diff
@@ -0,0 +1,49 @@
+package de.medizininformatikinitiative.torch.cql;
+
+import lombok.extern.slf4j.Slf4j;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.ListResource;
+import org.hl7.fhir.r4.model.MeasureReport;
+import org.hl7.fhir.r4.model.Parameters;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+
+@Slf4j
+public class CqlClient {
+    private final FhirConnector fhirConnector;
+    private final FhirHelper fhirHelper;
+
+    public CqlClient(FhirConnector fhirConnector,
+                     FhirHelper fhirHelper) {
+        this.fhirConnector = Objects.requireNonNull(fhirConnector);
+        this.fhirHelper = fhirHelper;
+    }
+
+
+    public Mono<List<String>> getPatientListByCql(String cqlQuery) {
+
+        var libraryUri = "urn:uuid" + UUID.randomUUID();
+        var measureUri = "urn:uuid" + UUID.randomUUID();
+        MeasureReport measureReport;
+
+        try {
+            Bundle bundle = fhirHelper.createBundle(cqlQuery, libraryUri, measureUri);
+            fhirConnector.transmitBundle(bundle);
+
+            Parameters params = fhirHelper.getListExecutionParams();
+            params.setParameter("measure", measureUri);
+            measureReport = fhirConnector.evaluateMeasure(measureUri, params);
+
+            var subjectListId = measureReport.getGroupFirstRep().getPopulationFirstRep().getSubjectResults().getReferenceElement().getIdPart();
+            return Mono.just(fhirConnector.searchAndExtractIds(subjectListId));
```

Comment:
a Mono isn't really useful here, if you just do everything blocking and return a Mono.just at the end